File: day2/flipping-the-matrix/solution.py
matrix = [
  [112, 42, 83, 119],
  [56, 125, 56, 49],
  [15, 78, 101, 43],
  [62, 98, 114, 108]
];
# matrix = [
#   [1, 2],
#   [3, 4]
# ];
# matrix = [
#   [1, 2, 3, 4],
#   [5, 6, 7, 8],
#   [9, 10, 11, 12],
#   [13, 14, 15, 16]
# ];

def flipping_the_matrix(matrix):
    last = int(len(matrix) -1)
    size = int(len(matrix)) // 2
    maximun = 0

    for i in range(0,size ):
        
        for j in range(0,size):
           
            maximun += max(matrix[i][j], matrix[i][last - j], matrix[last - i][j], matrix[last - i][last - j])
            
    
    
    return maximun

print(flipping_the_matrix(matrix))        

File: day2/flipping-the-matrix/solution.js
// let matrix = [
//   [112, 42, 83, 119],
//   [56, 125, 56, 49],
//   [15, 78, 101, 43],
//   [62, 98, 114, 108]
// ];
// let matrix = [
//   [1, 2],
//   [3, 4]
// ];
let matrix = [
  [1, 2, 3, 4],
  [5, 6, 7, 8],
  [9, 10, 11, 12],
  [13, 14, 15, 16]
];

// console.log(matrix[matrix.length - 1][0]);

function flippingTheMatrix(matrix) {
  let size = Math.floor(matrix.length / 2);
  let last = matrix.length - 1;
  let max = 0;

  for (let i = 0; i < size; i++) {
    for (let j = 0; j < size; j++) {
      max += Math.max(
        matrix[i][j],
        matrix[i][last - j],
        matrix[last - i][j],
        matrix[last - i][last - j]
      );
    }
  }
  return max;
}

console.log(flippingTheMatrix(matrix));
